workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -29,48 +42,116 @@ protected override void OnCreateManager()
-            var createPlayerRequest = new PlayerCreator.CreatePlayer.Request(playerCreatorEntityId, request);
-            commandSystem.SendCommand(createPlayerRequest);
+            playerRequestQueued = true;
+        }
+
+        private void SendCreatePlayerRequest()
+        {
+            commandSystem.SendCommand(new PlayerCreator.CreatePlayer.Request(
+                playerCreatorEntityIds[Random.Range(0, playerCreatorEntityIds.Count)],
+                new CreatePlayerRequest(serializedArgumentsCache)
+            ));
+            playerRequestQueued = false;
         }
 
         private void RetryCreatePlayerRequest()
         {
-            RequestPlayerCreation(serializedArgumentsCache);
+            if (playerCreationAttempts < PlayerLifecycleConfig.MaxPlayerCreationAttempts)
+            {
+                ++playerCreationAttempts;
+
+                logDispatcher.HandleLog(LogType.Warning, new LogEvent(
+                    $"Retrying player creation request, attempt {playerCreationAttempts}."
+                ));
+
+                SendCreatePlayerRequest();
+            }
+            else
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent(
+                    $"Unable to create player after {playerCreationAttempts} attempts."
+                ));
+            }
         }
 
         protected override void OnUpdate()
         {
-            if (PlayerLifecycleConfig.AutoRequestPlayerCreation && !initializationGroup.IsEmptyIgnoreFilter)
+            if (playerCreatorEntityIds.Count > 0)
             {
-                RequestPlayerCreation();
-            }
+                if (playerRequestQueued)
+                {
+                    SendCreatePlayerRequest();
+                }
 
-            // Currently this has a race condition where you can receive two entities
-            // The fix for this is more sophisticated server side handling of requests
-            var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
+                // Currently this has a race condition where you can receive two entities
+                // The fix for this is more sophisticated server side handling of requests
+                var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
 
-            for (var i = 0; i < responses.Count; i++)
-            {
-                ref readonly var response = ref responses[i];
-                if (response.StatusCode == StatusCode.AuthorityLost)
+                for (var i = 0; i < responses.Count; i++)
                 {
-                    RetryCreatePlayerRequest();
+                    ref readonly var response = ref responses[i];
+                    if (response.StatusCode == StatusCode.AuthorityLost)
+                    {
+                        RetryCreatePlayerRequest();
+                    }
+                    else if (response.StatusCode != StatusCode.Success)
```

Comment:
Surely a timeout status code warrants a retry? I'd recommend expanding this into a switch-case and grouping retry-able failures into a block that fall through into the Retry invocation.